health/health.d/wmi.conf

```diff
@@ -137,3 +137,175 @@ component: Disk
+       to: sysadmin
+
+ template: wmi_1m_ip_tcp_resets
+       on: wmi.tcp_conns_resets
+    class: Errors
+     type: Windows
+component: Network
+       os: *
+    hosts: *
+   lookup: average -1m at -10s unaligned absolute of ipv4,ipv6
+    units: tcp resets/s
+    every: 10s
+     info: average number of TCP RESETS over the last minute
+
+ template: wmi_10s_ip_tcp_resets
+       on: wmi.tcp_conns_active
+    class: Errors
+     type: System
+component: Network
+       os: *
+    hosts: *
+   lookup: average -10s unaligned absolute of ipv4,ipv6
+    units: tcp resets/s
+    every: 10s
+     warn: $netdata.uptime.uptime > (1 * 60) AND $this > ((($wmi_1m_ip_tcp_resets < 5)?(5):($wmi_1m_ip_tcp_resets)) * (($status >= $WARNING)  ? (1) : (10)))
+    delay: up 20s down 60m multiplier 1.2 max 2h
+  options: no-clear-notification
+     info: average number of sent TCP RESETS over the last 10 seconds. \
+           This can indicate a port scan, \
+           or that a service running on this host has crashed. \
+           Netdata will not send a clear notification for this alarm.
+       to: sysadmin
+
+
+### IIS
+# Source: https://learn.microsoft.com/en-us/answers/questions/495615/how-to-set-to-allow-maximumtcp-connection-in-windo.html
+# We have an alert to notify when TCP ports are exausted, but conggestion can conduct serves for another issue.
+ template: wmi_iis_congestion
+       on: wmi.iis_website_active_connections_count
+    class: Utilization
+     type: Windows
+component: IIS
+       os: *
+    hosts: *
+   lookup: sum -10s of active
+    units: connections
+    every: 10s
+     warn: $this > (($status >= $WARNING)  ? (375) : (325))
+     crit: $this > (($status == $CRITICAL) ? (425) : (375))
+    delay: up 20s down 20m multiplier 1.2 max 2h
+     info: You are having a higher number of active connections for $label:website, \
+           depending of hardware and configuration you can reach TCP congestion.
+       to: sysadmin
+
+### MSSQL
+# Source: https://www.aimbetter.com/cache-hit-ratio/
+ template: wmi_mssql_hit_ratio
+       on: wmi.mssql_instance_cache_hit_ratio
+    class: Workload
+     type: Windows
+component: MSSQL
+       os: *
+    hosts: *
+   lookup: average -1m unaligned of hit_ratio
+    units: %
+    every: 10s
+     warn: $this < (($status >= $WARNING)  ? (80) : (70))
+     crit: $this < (($status == $CRITICAL) ? (69) : (50))
+    delay: down 15m multiplier 1.5 max 1h
+     info: Average cache hit ratio in db $label:mssql_instance over the last minute
+       to: sysadmin
+
+ template: wmi_mssql_blocked_processes
+       on: wmi.mssql_instance_blocked_processes
+    class: Workload
+     type: Windows
+component: MSSQL
+       os: *
+    hosts: *
+   lookup: average -30s unaligned of blocked
+    units: %
+    every: 10s
+     warn: $this > (($status >= $WARNING)  ? (25) : (30))
+     crit: $this > (($status == $CRITICAL) ? (40) : (50))
```

Comment:
Where did you get those numbers? Why 30 and 50?

Reply:
Why should I have a bigger value? Processes are blocked and query are not running in 30 seconds, the server is overloaded (probably a lot of `select * from TABLE` instead to specify fields).

A direct answer is experience with PostgreSQL/MariaDB/Oracle. We never wanted a lot of queries in queue.